src/Database/PG/Query/Class.hs

```diff
@@ -75,12 +79,26 @@ newtype SingleRow a = SingleRow
 type AltJ :: Type -> Type
 newtype AltJ a = AltJ {getAltJ :: a}
 
-mapEither :: (a -> c) -> (b -> d) -> Either a b -> Either c d
-mapEither f _ (Left a) = Left $ f a
-mapEither _ f (Right b) = Right $ f b
-
 instance (FromJSON a) => FromCol (AltJ a) where
-  fromCol = fromColHelper (PD.fn $ mapEither fromString AltJ . eitherDecodeStrict)
+  fromCol =
+    decodeJson >=> parse
```

Comment:
This implementation is very confusing to me. Why do we have to parse the `ByteString` into `Value` first and then into `a`? 

Isn't this much simpler? 

```haskell
fromCol = fromColHelper (PD.fn $ bimap fromString AltJ . eitherDecodeStrict . dropFirst)
```

Reply:
Good point - that still passes all the tests, so implemented here: https://github.com/hasura/pg-client-hs/pull/76

And testing performance in HGE here: https://github.com/hasura/graphql-engine-mono/pull/5709